### PR TITLE
Add bob.learn.boosting recipe

### PR DESCRIPTION
**Pull request by 183amir**

### CODE REVIEW

File: recipes/bob.learn.boosting/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - boosting_example.py = bob.learn.boosting.examples.mnist:main
+  number: 0
+  skip: true  # [not linux]
+  script: python -B setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+  - python
+  - setuptools
+  - scipy
```
Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: Would be nice to have `numpy`.

File: recipes/bob.learn.boosting/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  - bob.blitz
+  - bob.core
+  - bob.io.base
+  - cmake
+  - pkg-config
+  - gcc     # [linux]
+
+  run:
+  - python
+  - scipy
```
Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: Same.

**Comment by 183amir** [comment]
Is there a reason this is not being merged?

File: recipes/bob.learn.boosting/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+source:
+  fn: bob.learn.boosting-{{ version }}.zip
+  url: https://pypi.python.org/packages/source/b/bob.learn.boosting/bob.learn.boosting-{{ version }}.zip
+  md5: d02c5be0c59f55587409a558a98bf04e
+
+build:
+  entry_points:
+    - boosting_example.py = bob.learn.boosting.examples.mnist:main
+  number: 0
+  skip: true  # [not linux]
```
Comment (ocefpaf) [inline recipes/bob.learn.boosting/meta.yaml]: Any special reason for skipping OS X too?

File: recipes/bob.learn.boosting/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    - setuptools
+    - numpy
+    - scipy
+    - bob.extension
+    - bob.blitz
+    - bob.core
+    - bob.io.base
+    - cmake
+    - pkg-config
+    - gcc     # [linux]
```
Comment (ocefpaf) [inline recipes/bob.learn.boosting/meta.yaml]: Maybe it is worth adding a comment to why you need the default channel gcc.

File: recipes/bob.learn.boosting/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: I think we made one of its dependencies Linux only, but I'm not remembering right now.

Comment (183amir) [inline recipes/bob.learn.boosting/meta.yaml]: `bob.io.base` is blocking the `osx` builds.

Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: Thanks for clarifying.

File: recipes/bob.learn.boosting/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: I thought bob packages all need C++0x. Is that correct?

Comment (183amir) [inline recipes/bob.learn.boosting/meta.yaml]: I thought adding `gcc` for `linux` was the default way to go.

Comment (183amir) [inline recipes/bob.learn.boosting/meta.yaml]: > I thought bob packages all need C++0x. Is that correct?
Yes it was. I am confused now. Should I add `gcc` for Linux or not.

Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: So, there is a compiler on Linux so if you don't need this don't add it. Our plan is to make this unnecessary in the future. Even for C++11 support (possibly C++14 too). Though I thought we had a discussion with one of bob's other developers and determined we might need this. I don't remember which one it was one though. Maybe bob.math or something like that.

Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: Then keep it.

Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: Sorry hadn't seen your comment when I posted the previous one.

File: recipes/bob.learn.boosting/meta.yaml [hunk 5]
```diff
@@ -28,7 +28,6 @@ requirements:
     - bob.io.base
     - cmake
     - pkg-config
-    - gcc     # [linux]
```
Comment (jakirkham) [inline recipes/bob.learn.boosting/meta.yaml]: You can keep it. No need to remove it.

**Comment by 183amir** [comment]
Ok to make things simple, I will add `gcc` to all `bob` packages which contain `C++` code automatically. Or do we have to verify them one by one?

**Comment by jakirkham** [comment]
> Ok to make things simple, I will add `gcc` to all `bob` packages which contain `C++` code automatically. Or do we have to verify them one by one?

Agreed. Sorry for the confusion.

**Comment by 183amir** [comment]
So can we get this merged please?

**Comment by jakirkham** [comment]
LGTM. Final comments @ocefpaf?

**Comment by ocefpaf** [comment]
Nope. Let's merge. Thanks @183amir for the perseverance.  I never used `bob` but I am curious to try it!

**Comment by 183amir** [comment]
Thank you for this great channel and all the hard work you put in.
`bob` is very nice for biometric experiments. `idiap` is even selling commercial licenses for it now (for the GPLv3 licensed ones).

**Pull request merged**